### PR TITLE
Use cookie file for btsieve's bitcoind authentication

### PR DESCRIPTION
**Pull request by da-kami**
Automated test to be added once https://github.com/testcontainers/testcontainers-node/issues/20 is resolved. 

For now behaviour was tested manually by starting up btsieve with this config:

```
[bitcoin]
node_url = "http://localhost:18443"
zmq_endpoint = "tcp://127.0.0.1:28332"

[bitcoin.authentication.cookie]
file_path = "/Users/dakami/CoBloX/TEMP/cookiefile-test/config/.cookie"

[http_api]
address_bind="0.0.0.0"
port_bind=8181

[log_levels]
btsieve="DEBUG"
```

against this container:
```
version: "3"
services:
    btc:
        image: coblox/bitcoin-core:0.17.0
        volumes:
            - "./config:/root/.bitcoin/regtest"
        command: -regtest
            -server
            -printtoconsole
            -rpcbind=0.0.0.0:18443
            -rpcallowip=0.0.0.0/0
            -debug=1
            -zmqpubrawblock=tcp://*:28332
            -zmqpubrawtx=tcp://*:28333
            -acceptnonstdtxn=0
        ports:
            - "18443:18443"
            - "28332:28332"
            - "28333:28333"
```

Note: when changing the password in the mapped `.cookie` file manually btsieve fails with "Unautherized" as expected.

### CODE REVIEW

File: btsieve/config/bitcoin_basicauth.toml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+[bitcoin]
```
Comment (D4nte) [inline btsieve/config/bitcoin_basicauth.toml]: We just discussed a testing strategy with @thomaseizinger involving multiple configuration files for `cnd`. I think it'd be good to do the same strategy for btsieve. Please chat with him.

File: btsieve/config/bitcoin_basicauth.toml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+node_url = "http://localhost:18443"
+zmq_endpoint = "tcp://127.0.0.1:28332"
+
+[bitcoin.authentication.basic]
+node_username = "Hans"
+node_password = "Wurst"
+
```
Comment (D4nte) [inline btsieve/config/bitcoin_basicauth.toml]:
```suggestion
```

File: btsieve/config/bitcoin_cookieauth.toml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+[bitcoin]
+node_url = "http://localhost:18443"
+zmq_endpoint = "tcp://127.0.0.1:28332"
+
+[bitcoin.authentication.cookie]
+file_path = "/home/bitcoin/.bitcoin/regtest/.cookie"
+
```
Comment (D4nte) [inline btsieve/config/bitcoin_cookieauth.toml]:
```suggestion
```

File: btsieve/config/bitcoin_only.toml [hunk 4]
```diff
@@ -1,8 +1,10 @@
 [bitcoin]
 node_url = "http://localhost:18443"
+zmq_endpoint = "tcp://127.0.0.1:28332"
+
+[bitcoin.authentication.basic]
 node_username = "bitcoin"
 node_password = "54pLR_f7-G6is32LP-7nbhzZSbJs_2zSATtZV_r05yg="
-zmq_endpoint = "tcp://127.0.0.1:28332"
```
Comment (D4nte) [inline btsieve/config/bitcoin_only.toml]:
```suggestion
```

File: btsieve/src/settings/mod.rs [hunk 5]
```diff
@@ -124,4 +153,37 @@ mod tests {
...
+            .map(|b| &b.authentication)
+            .is_equal_to(cookie_authentication);
+    }
+
+    #[test]
+    fn can_read_config_with_bitcoin_basic_authentication() {
+        let settings = Settings::read("./config/bitcoin_basicauth.toml");
+
+        let basic_authentication = BitcoinAuth::Basic {
+            node_username: "Hans".to_owned(),
```
Comment (bonomat) [inline btsieve/src/settings/mod.rs]: 🙃
```suggestion
            node_username: "Satoshi".to_owned(),
```

File: btsieve/src/settings/mod.rs [hunk 6]
```diff
@@ -124,4 +153,37 @@ mod tests {
...
+            node_password: "Wurst".to_owned(),
```
Comment (bonomat) [inline btsieve/src/settings/mod.rs]: 🙃
```suggestion
            node_password: "Nakamoto".to_owned(),
```

File: btsieve/config/bitcoin_basicauth.toml [hunk 7]
```diff
@@ -0,0 +1,13 @@
+[bitcoin]
+node_url = "http://localhost:18443"
+zmq_endpoint = "tcp://127.0.0.1:28332"
+
+[bitcoin.authentication.basic]
+node_username = "Hans"
```
Comment (bonomat) [inline btsieve/config/bitcoin_basicauth.toml]: 🙃
```suggestion
node_username = "Satoshi"
```

File: btsieve/config/bitcoin_basicauth.toml [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+node_password = "Wurst"
```
Comment (bonomat) [inline btsieve/config/bitcoin_basicauth.toml]: 🙃
```suggestion
node_password = "Nakamoto"
```

**Comment by thomaseizinger** [comment]
@da-kami Fixed it up for you by squashing all those commits into a single one :)

File: btsieve/config/bitcoin_only.toml (hunk 4 above)
Comment (da-kami) [reply btsieve/config/bitcoin_only.toml]: I don't understand what you want me to do here :)

File: btsieve/config/bitcoin_basicauth.toml (hunk 1 above)
Comment (da-kami) [reply btsieve/config/bitcoin_basicauth.toml]: Conclusion after discussion with @thomaseizinger: should be done separate from this issue.

File: btsieve/config/bitcoin_only.toml (hunk 4 above)
Comment (D4nte) [reply btsieve/config/bitcoin_only.toml]: Remove the double newline

Comment (da-kami) [reply btsieve/config/bitcoin_only.toml]: removed manually.

**Pull request merged**